gen/weighted_test.go

```diff
@@ -0,0 +1,41 @@
+package gen_test
+
+import (
+	"testing"
+
+	"github.com/leanovate/gopter/gen"
+)
+
+func TestWeighted(t *testing.T) {
+	weighted := gen.Weighted([]gen.WeightedGen{
+		{1, gen.Const("A")},
+		{2, gen.Const("B")},
+		{7, gen.Const("C")},
+	})
+	results := make(map[string]int)
+	for i := int64(0); i < int64(1000); i++ {
+		result, ok := weighted.Sample()
```

Comment:
In most of the tests I do a "gen(fixedParameters).Retrieve()" where fixedParameters give a rng with fixed seed, so that I get reproducable outputs.
The delta-approach is of course valid as well.